src/_imaging.c

```diff
@@ -4087,15 +4117,21 @@ _set_blocks_max(PyObject *self, PyObject *args) {
     }
 
     if ((unsigned long)blocks_max >
-        SIZE_MAX / sizeof(ImagingDefaultArena.blocks_pool[0])) {
+        SIZE_MAX / sizeof(ImagingGetArena()->blocks_pool[0])) {
         PyErr_SetString(PyExc_ValueError, "blocks_max is too large");
         return NULL;
     }
 
-    MUTEX_LOCK(&ImagingDefaultArena.mutex);
-    int status = ImagingMemorySetBlocksMax(&ImagingDefaultArena, blocks_max);
-    MUTEX_UNLOCK(&ImagingDefaultArena.mutex);
-    if (!status) {
+    int status = 0;
+    uint8_t index;
+    ImagingMemoryArena arena;
+    IMAGING_ARENAS_FOREACH(index, arena) {
+        MUTEX_LOCK(&arena->mutex);
+        status |= ImagingMemorySetBlocksMax(arena, blocks_max);
```

Comment:
Maybe change the name of the variable to something like `error` here, since we've changed the return value?

Reply:
👍